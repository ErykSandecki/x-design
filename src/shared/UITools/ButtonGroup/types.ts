// types
import { TTooltipProps } from '../../UI/Tooltip/Tooltip';
import { TIconProps } from '../../UI/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
