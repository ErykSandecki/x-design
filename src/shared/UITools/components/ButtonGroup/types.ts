// types
import { TTooltipProps } from '../../../UI/components/Tooltip/Tooltip';
import { TIconProps } from '../../../UI/components/Icon/Icon';

export type TButtonGroup = {
  disabled?: boolean;
  name: TIconProps['name'];
  onClick: TFunc;
  tooltip?: Omit<TTooltipProps, 'children'>;
};
