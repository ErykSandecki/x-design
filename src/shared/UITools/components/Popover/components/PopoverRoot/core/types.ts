export type TContext = {
  setSelected: TFunc<[boolean]>;
};
