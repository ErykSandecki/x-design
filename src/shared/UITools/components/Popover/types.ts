// types
import { TIconProps } from 'shared/UI/components/Icon/Icon';

export type TPopoverData = {
  icon?: TIconProps['name'];
  onClick?: () => void;
  selected?: boolean;
  text: string;
};

export type TPopoverSeparator = { separator: boolean };

export type TPopoverItem = TPopoverData | TPopoverSeparator;

export type TPopover = {
  data: Array<TPopoverData | TPopoverSeparator>;
};
