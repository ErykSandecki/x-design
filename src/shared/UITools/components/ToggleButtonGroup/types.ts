import { TIconProps } from 'shared/UI/components/Icon/Icon';
import { TToggleButtonProps } from './components/ToggleButton/ToggleButton';

export type TToggleButton<V> = Pick<TToggleButtonProps<V>, 'tooltip' | 'value'> & {
  icon: TIconProps['name'];
};

export type TToggleButtonGroupValue = string | Array<string>;
