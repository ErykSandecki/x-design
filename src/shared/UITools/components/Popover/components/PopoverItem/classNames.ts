export const className = 'PopoverItem';

export const classNames = {
  [className]: className,
  checkIcon: {
    name: `${className}__check-icon`,
    modificators: {
      selected: `${className}__check-icon--selected`,
    },
  },
  icon: `${className}__icon`,
  text: `${className}__text`,
};
