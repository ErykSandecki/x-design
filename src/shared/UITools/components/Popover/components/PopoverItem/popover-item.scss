.PopoverItem {
  @include theme-selector('blue-2', background-color, '&:hover');

  align-items: center;
  border-radius: 5px;
  column-gap: 10px;
  cursor: pointer;
  display: flex;
  height: 24px;
  justify-content: left;
  padding: 0 5px;

  &__check-icon {
    flex-shrink: 0;
    margin-right: 2.5px;
    opacity: 0;

    &--selected {
      opacity: 1;
    }
  }

  &__icon {
    flex-shrink: 0;
  }

  &__text {
    flex-shrink: 0;
    font-size: 12px;
  }
}
