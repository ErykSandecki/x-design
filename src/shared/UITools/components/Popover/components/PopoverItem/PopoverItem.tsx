import { FC } from 'react';

// components
import E2EDataAttribute from '../../../../../E2EDataAttributes/E2EDataAttribute';
import { Box, Icon, Small, TIconProps } from 'shared';

// core
import { usePopoverRoot } from '../PopoverRoot/core/PopoverRootProvider';

// hooks
import { useClickEvent } from './hooks/useClickEvent';
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './popover-item.scss';

// types
import { E2EAttribute } from 'types';

export type TPopoverItemProps = {
  icon?: TIconProps['name'];
  index: number;
  onClick?: TFunc;
  selected?: boolean;
  text: string;
};

export const PopoverItem: FC<TPopoverItemProps> = ({ icon, index, onClick, selected, text }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { setSelected } = usePopoverRoot();
  const onClickHandler = useClickEvent(onClick, setSelected);

  return (
    <E2EDataAttribute type={E2EAttribute.popoverItem} value={index}>
      <Box
        classes={{ className: cx(classNamesWithTheme[className]) }}
        onClick={onClickHandler}
        sx={{
          alignItems: 'center',
          borderRadius: '5px',
          columnGap: '10px',
          display: 'flex',
          height: '24px',
          justifyContent: 'left',
          px: 5,
        }}
      >
        <Icon
          classes={{
            className: cx(classNamesWithTheme.checkIcon.name, [
              classNamesWithTheme.checkIcon.modificators.selected,
              selected,
            ]),
          }}
          height={12}
          name="Check"
          width={12}
        />
        {icon && <Icon classes={{ className: cx(classNamesWithTheme.icon) }} height={12} name={icon} width={12} />}
        <Small classes={{ className: cx(classNamesWithTheme.text) }}>{text}</Small>
      </Box>
    </E2EDataAttribute>
  );
};

export default PopoverItem;
