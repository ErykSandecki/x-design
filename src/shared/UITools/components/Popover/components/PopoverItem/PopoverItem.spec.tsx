import { fireEvent, render } from '@testing-library/react';

// components
import PopoverItem from './PopoverItem';

// types
import { E2EAttribute } from 'types';

// utils
import { getByE2EAttribute } from 'test';

const mockCallBack = jest.fn();

describe('PopoverItem snapshots', () => {
  it('should render PopoverItem', () => {
    // before
    const { asFragment } = render(
      <PopoverItem
        icon="AlignHorizontalCenter"
        index={0}
        onClick={mockCallBack}
        selected={false}
        setSelected={mockCallBack}
        text="text"
      />,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render selected', () => {
    // before
    const { asFragment } = render(
      <PopoverItem
        icon="AlignHorizontalCenter"
        index={0}
        onClick={mockCallBack}
        selected
        setSelected={mockCallBack}
        text="text"
      />,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('PopoverItem behaviors', () => {
  it('should click item', () => {
    // before
    const { container } = render(
      <PopoverItem
        icon="AlignHorizontalCenter"
        index={0}
        onClick={mockCallBack}
        selected={false}
        setSelected={mockCallBack}
        text="text"
      />,
    );

    // find
    const item = getByE2EAttribute(container, E2EAttribute.popoverItem, 0);

    // action
    fireEvent.click(item);

    // result
    expect(mockCallBack.mock.calls[1][0]).toBe(false);
  });
});
