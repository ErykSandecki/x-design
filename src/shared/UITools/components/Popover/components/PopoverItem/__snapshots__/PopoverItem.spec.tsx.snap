// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopoverItem snapshots should render PopoverItem 1`] = `
<DocumentFragment>
  <div
    class="PopoverItem PopoverItem--light"
    data-test-popover-item="0"
  >
    <div
      class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="check"
      height="12"
      name="Check"
      width="12"
    />
    <div
      class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="align-horizontal-center"
      height="12"
      name="AlignHorizontalCenter"
      width="12"
    />
    <small
      class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      text
    </small>
  </div>
</DocumentFragment>
`;

exports[`PopoverItem snapshots should render selected 1`] = `
<DocumentFragment>
  <div
    class="PopoverItem PopoverItem--light"
    data-test-popover-item="0"
  >
    <div
      class="css-1jnxvbp PopoverItem__checkIcon PopoverItem__checkIcon--light PopoverItem__checkIcon--selected PopoverItem__checkIcon--selected--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="check"
      height="12"
      name="Check"
      width="12"
    />
    <div
      class="css-1jnxvbp PopoverItem__icon PopoverItem__icon--light Icon Icon--light Icon--neutral1 Icon--neutral1--light"
      data-test-icon="align-horizontal-center"
      height="12"
      name="AlignHorizontalCenter"
      width="12"
    />
    <small
      class="css-1jnxvbp PopoverItem__text PopoverItem__text--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
      data-test-text=""
      style="text-align: inherit;"
    >
      text
    </small>
  </div>
</DocumentFragment>
`;
